src/collapse/useCollapse.ts

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from 'react';
+import { GraphEdge, GraphNode } from 'types';
+import { getExpandPath, getVisibleEntities } from './collapse';
+
+export interface UseCollapseProps {
+  /**
+   * Current collapsed node ids.
+   */
+  collapsedNodeIds?: string[];
+
+  /**
+   * Node data.
+   */
+  nodes?: GraphNode[];
+
+  /**
+   * Edge data.
+   */
+  edges?: GraphEdge[];
+}
+
+export interface CollpaseResult {
+  /**
+   * Determine if a node is currently collapsed
+   */
+  isCollapsed: (nodeId: string) => boolean;
+
+  /**
+   * Return a list of ids required to expand in order to view the provided node
+   */
+  getExpandPathIds: (nodeId: string) => string[];
+}
+
+export const useCollapse = ({
+  collapsedNodeIds = [],
+  nodes = [],
+  edges = []
+}: UseCollapseProps): CollpaseResult => {
+  const isCollapsed = useCallback(
+    (nodeId: string) => {
+      const { visibleNodes } = getVisibleEntities({
+        nodes,
+        edges,
+        collapsedIds: collapsedNodeIds
+      });
+      const visibleNodeIds = visibleNodes.map(n => n.id);
+
+      return visibleNodeIds.includes(nodeId);
+    },
+    [collapsedNodeIds, edges, nodes]
+  );
+
+  const getExpandPathIds = useCallback(
+    (nodeId: string) => {
+      const { visibleEdges } = getVisibleEntities({
+        nodes,
+        edges,
+        collapsedIds: collapsedNodeIds
+      });
+      const visibleEdgeIds = visibleEdges.map(e => e.id);
+
+      return getExpandPath({ nodeId, edges, visibleEdgeIds });
+    },
+    [collapsedNodeIds, edges, nodes]
+  );
+
+  return {
+    isCollapsed,
```

Comment:
This name makes it sound like its a boolean not a function, it should be `getIsCollapsed`.